docs/learning/backend/PullMerge.md

```diff
@@ -0,0 +1,148 @@
+# Pull merge & conflict resolution
+
+> NOTE This document is work in progress and Rebase API are **experimental**.
+
+## What is change merging?
+
+Change merging is when a briefcase pulls new changes from the iModel Hub and applies them locally. If the user has any local changes, the incoming changes need to be merged with the local changes. In this document, we are currently strictly talking about low-level SQLite changesets and high-level concepts defined in ECSchemas.
+
+## What is change conflicts
+
+Git is used for text and text is two-dimensional data. Conflicts in two-dimensional data happen when two users change overlapping ranges of text. In a database, an overlapping change can be a change to the same row identified by a primary key. However, it could be more complex as that row might have database constraints like unique index, check, foreign key, triggers, etc., that make this row part of a larger chunk of information. SQLite tracks changes to individual tables/rows. It is up to the application to make changes consistently so as not to violate database constraints. If the database only contains a single table with no constraints, then it's basically the same as a text file as long as there is a primary key.
+
+SQLite records changes by recording INSERT/UPDATE/DELETE operations with old and new values into a binary changeset. The order of changes in the changeset is arbitrary, but once the whole changeset is applied, it should bring the database to a new state that is consistent with all specified database constraints.
+
+Things get interesting when there are local changes specifically to the same data as the incoming changeset. This operation is called pull-merge. Doing that can result in conflicts which from the SQLite perspective is as follows.
+
+|  | Operation | Conflict description                                                    | Type         |
+|--|-----------|-------------------------------------------------------------------------|--------------|
+|  | `INSERT`  | 1. PRIMARY KEY already exists                                           | `Conflict`   |
+|  | `DELETE`  | 2. PRIMARY KEY does not exist                                           | `NotFound`   |
+|  | `UPDATE`  | 3. PRIMARY KEY does not exist                                           | `NotFound`   |
+|  |           | 1. Database constraint voliation e.g. UNIQUE, CHECK                     | `Constraint` |
+|  |           | 2. PRIMARY KEY does not exist but other fields values does not match    | `Data`       |
+|  |           | 3. PRIMARY KEY does not exist but other fields values does not match.   | `Data`       |
+|  |           | 4. Database constraint voliation e.g. UNIQUE, CHECK                     | `Constraint` |
```

Comment:
Duplicate from 3 rows above. I think maybe you meant to put different operation names in the first column?